cmd/virt-xml-hook/virt_xml_hook.go

```diff
@@ -0,0 +1,139 @@
+
+	return &hooksInfo.InfoResult{
+		Name: "update-xml",
+		Versions: []string{
+			s.Version,
+		},
+		HookPoints: []*hooksInfo.HookPoint{
+			{
+				Name:     hooksInfo.OnDefineDomainHookPointName,
+				Priority: 0,
+			},
+		},
+	}, nil
+}
+
+const virtXML = "virt-xml"
+
+type v1alpha1Server struct {
+	args []string
+}
+
+type v1alpha2Server struct {
+	args []string
+}
+
+func MergeKubeVirtXMLWithProvidedXML(domainXML []byte, args []string) ([]byte, error) {
+	args = append(args, "--edit")
+	args = append(args, "--print-xml")
+	cmd := exec.Command(virtXML, args...)
+	cmd.Stdin = strings.NewReader(string(domainXML))
+	var stderr bytes.Buffer
+	cmd.Stderr = &stderr
+	log.Log.Infof("Execute command: %s", cmd.String())
+	out, err := cmd.Output()
+	if err != nil {
+		log.Log.Errorf("Fail running command stdout:%s stderr: %s error:%v", out, stderr.String(), err)
+		return []byte{}, err
+	}
+
+	return []byte(out), nil
+}
+
+func (s v1alpha2Server) OnDefineDomain(ctx context.Context, params *hooksV1alpha2.OnDefineDomainParams) (*hooksV1alpha2.OnDefineDomainResult, error) {
+	log.Log.Info(onDefineDomainLoggingMessage)
+	newDomainXML, err := MergeKubeVirtXMLWithProvidedXML(params.GetDomainXML(), s.args)
+	if err != nil {
+		return nil, err
+	}
+	return &hooksV1alpha2.OnDefineDomainResult{
+		DomainXML: newDomainXML,
+	}, nil
+}
+func (s v1alpha2Server) PreCloudInitIso(_ context.Context, params *hooksV1alpha2.PreCloudInitIsoParams) (*hooksV1alpha2.PreCloudInitIsoResult, error) {
+	return &hooksV1alpha2.PreCloudInitIsoResult{
+		CloudInitData: params.GetCloudInitData(),
+	}, nil
+}
+
+func (s v1alpha1Server) OnDefineDomain(ctx context.Context, params *hooksV1alpha1.OnDefineDomainParams) (*hooksV1alpha1.OnDefineDomainResult, error) {
+	log.Log.Info(onDefineDomainLoggingMessage)
+	newDomainXML, err := MergeKubeVirtXMLWithProvidedXML(params.GetVmi(), s.args)
+	if err != nil {
+		return nil, err
+	}
+	return &hooksV1alpha1.OnDefineDomainResult{
+		DomainXML: newDomainXML,
+	}, nil
+}
+
+func main() {
+	log.InitializeLogging("xml update")
+
+	var version string
+	pflag.StringVar(&version, "version", "", "hook version to use")
+
+	var options string
+	pflag.StringVar(&options, "args", "", "params to pass to virt-xml")
+	pflag.Parse()
+
+	args := strings.Split(options, "|")
+
+	socketPath := hooks.HookSocketsSharedDirectory + "/update.sock"
+	socket, err := net.Listen("unix", socketPath)
+	if err != nil {
+		log.Log.Reason(err).Errorf("Failed to initialized socket on path: %s", socket)
+		log.Log.Error("Check whether given directory exists and socket name is not already taken by other file")
+		panic(err)
+	}
+	defer os.Remove(socketPath)
+
+	server := grpc.NewServer([]grpc.ServerOption{}...)
+
+	if version == "" {
+		panic(fmt.Errorf(usage))
+	}
+	if options == "" {
+		panic(fmt.Errorf(usage))
+	}
+
+	hooksInfo.RegisterInfoServer(server, infoServer{Version: version})
+	hooksV1alpha1.RegisterCallbacksServer(server, v1alpha1Server{args: args})
```

Comment:
Not sure, the [smbios example](https://github.com/kubevirt/kubevirt/blob/main/cmd/example-hook-sidecar/smbios.go#L168-L169) does

Reply:
I think that might be because that specific hook was introduced before the v1alpha2 API existed, and was later extended in a way that retained backwards compatibility. I don't think we need to care in our case.